include/highfive/H5DataSpace.hpp

```diff
@@ -107,51 +107,13 @@ class DataSpace : public Object {
     /// associated dataset maximum dimension
     std::vector<size_t> getMaxDimensions() const;
 
-    /// Create a dataspace matching a single element of a basic type
-    ///  supported type are integrals (int,long), floating points (float,double)
-    ///  and std::string
-    template <typename ScalarValue>
-    static DataSpace From(const ScalarValue& scalar_value);
-
-    /// Create a dataspace matching the container dimensions and size
-    /// Supported Containers are:
-    ///  - vector of fundamental types
-    ///  - vector of std::string
-    ///  - boost::multi_array (with H5_USE_BOOST defined)
-    template <typename Value>
-    static DataSpace From(const std::vector<Value>& container);
-
-    /// Create a dataspace matching the container dimensions for a
-    /// std::array.
-    template <typename Value, std::size_t N>
-    static DataSpace From(const std::array<Value, N>&);
-
-    template <typename ValueT, std::size_t N>
-    static DataSpace From(const ValueT(&container)[N]);
+    /// Create a dataspace matching a type accepted by details::inspector
+    template <typename T>
+    static DataSpace From(const T& value);
 
     template <std::size_t N, std::size_t Width>
     static DataSpace FromCharArrayStrings(const char(&)[N][Width]);
```

Comment:
Could this be added to the generic template as well?

Reply:
not now, ArrayStrings are a bit complicate.